graphql_compiler/compiler/compiler_frontend.py

```diff
@@ -82,6 +82,13 @@
                       is_vertex_field_name, strip_non_null_from_type, validate_safe_string)
 
 
+# LocationStackEntry contains the following:
+# - location: Location object correspoding to an inserted MarkLocation block
+# - num_traverses: Int counter for the number of traverses intserted after the last MarkLocation
+#                  (corresponding Location stored in `location`)
+LocationStackEntry = namedtuple('LocationStackEntry', ('location', 'num_traverses'))
```

Comment:
Bonus points if you make a helper function that makes these objects after asserting that `num_traverses` is non-negative and that `location` is a `Location` object. This helper would be the way to construct such objects everywhere, instead of explicitly calling the constructor or using `._replace()`